core/src/main/java/io/seata/core/rpc/netty/NettyClientChannelManager.java

```diff
@@ -181,12 +181,21 @@ void reconnect(List<String> availList, String transactionServiceGroup) {
             }
             if (failedMap.size() > 0) {
                 if (LOGGER.isInfoEnabled()) {
-                    LOGGER.error("{} can not connect to {} cause:{}", FrameworkErrorCode.NetConnect.getErrCode(), failedMap.keySet(), failedMap.values().stream().map(Throwable::getMessage).collect(Collectors.toSet()));
+                    LOGGER.error("{} can not connect to {} cause:{}", FrameworkErrorCode.NetConnect.getErrCode(),
+                            failedMap.keySet(),
+                            failedMap.values().stream().map(Throwable::getMessage).collect(Collectors.toSet()));
                 } else if (LOGGER.isDebugEnabled()) {
                     failedMap.forEach((key, value) -> {
-                        LOGGER.error("{} can not connect to {} cause:{} trace information:{}", FrameworkErrorCode.NetConnect.getErrCode(), key, value.getMessage(), value);
+                        LOGGER.error("{} can not connect to {} cause:{} trace information:{}",
+                                FrameworkErrorCode.NetConnect.getErrCode(), key, value.getMessage(), value);
                     });
                 }
+                boolean failFast = NettyClientConfig.isEnableClientChannelCheckFailFast();
+                boolean main = Thread.currentThread().getId() == 1L;
```

Comment:
Is threadID=1 always the main/initial thread ?

Consider an external Tomcat container. Is the application running on the main thread, and is the thread ID 1?